src/editor/FootnoteEditor.jsx

```diff
@@ -0,0 +1,147 @@
+import { isEqual } from 'lodash';
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import { ReactEditor } from 'slate-react';
+import { setPluginOptions } from 'volto-slate/actions';
+import { Icon as VoltoIcon } from '@plone/volto/components';
+import briefcaseSVG from '@plone/volto/icons/briefcase.svg';
+import checkSVG from '@plone/volto/icons/check.svg';
+import clearSVG from '@plone/volto/icons/clear.svg';
+import InlineForm from './InlineForm';
+import { Node } from 'slate';
+import { getAllBlocks } from 'volto-slate/utils';
+
+const FootnoteEditor = (props) => {
+  const {
+    editor,
+    schemaProvider,
+    pluginId,
+    getActiveElement,
+    isActiveElement,
+    insertElement,
+    unwrapElement,
+    hasValue,
+    onChangeValues,
+  } = props;
+
+  const dispatch = useDispatch();
+  const [formData, setFormData] = React.useState({});
+
+  const active = getActiveElement(editor);
+
+  if (!active) {
+    console.error('Active element not found, this will crash');
+  }
+  const [elementNode] = active;
+  const isElement = isActiveElement(editor);
+
+  const blocks = [];
+  const filteredBlocks = [];
+
+  const blockProps = editor.getBlockProps();
+  const metadata = blockProps.metadata || blockProps.properties;
+  getAllBlocks(metadata, blocks);
```

Comment:
This API is meh. Still, you can do `const blocks = getAllBlocks(metadata, [])` to make it less ugly

Reply:
I was thinking of refactoring getAllBlocks, to return value instead of modifying the argument, or I can just make the change